react/features/prejoin/functions.js

```diff
@@ -162,6 +162,6 @@ export function isPrejoinPageVisible(state: Object): boolean {
 export function shouldAutoKnock(state: Object): boolean {
     const { iAmRecorder, iAmSipGateway, autoKnockLobby } = state['features/base/config'];
 
-    return (isPrejoinPageVisible(state) || autoKnockLobby || (iAmRecorder && iAmSipGateway))
-        && !state['features/lobby'].knocking;
+    return (!isPrejoinPageVisible(state) || autoKnockLobby || (iAmRecorder && iAmSipGateway))
```

Comment:
isPrejoinVisible returns false, I guess that's ok because we are in lobby when auto knock triggers.
The rest of the values return undefined, but maybe they are useful in other cases.

Reply:
Not sure that's what we want. If the prejoin page was never visible we don't want to auto-knock.

Reply:
Even if displayName is set?

Reply:
Yep. If the prejoin page was not visible we should never auto-knock.